PWGGA/GammaConvBase/AliConversionPhotonCuts.cxx

```diff
@@ -3276,17 +3348,103 @@ Bool_t AliConversionPhotonCuts::SetSinglePtCut(Int_t singlePtCut){   // Set Cut
+    fSinglePtCut = 0.040;
+    fPtCut       = 0.075;
     break;
   case 27: // r: 0.036 GeV + min gamma pT cut of 20 MeV ; equiv. 0.09 for lowB
-    fSinglePtCut = 0.036;
-    fPtCut       = 0.01;
+    //A.M. redefinition 20.10.15. In sys. uncertainty I do not see any variation in efficiency for 8,7,p,6
+    fSinglePtCut = 0.04;
+    fPtCut       = 0.100;
     break;
   case 28: // s: 0.040 GeV + min gamma pT cut of 20 MeV ; equiv. 0.075 for lowB
     fSinglePtCut = 0.040;
     fPtCut       = 0.01;
     break;
+  case 29: //t  R-Dep pT cut NomB Default 0.050 GeV + min gamma pT cut of 100 MeV
+    fSinglePtCut = 0.050;
+    fPtCut       = 0.1;
+    fDoRDepPtCut = kTRUE;
+    //    SetPtCutArraySize(fPtCutArraySize);
+    fRArray = new Double_t[fPtCutArraySize];
+    fRArray[0] = 13.;
+    fRArray[1] = 33.5;
+    fRArray[2] = 55.;
+    fRArray[3] = 72.;
+    fRArray[4] = 95.;
+    fRArray[5] = 180.;    
+
+    fRDepPtCutArray = new Double_t[fPtCutArraySize];
+    fRDepPtCutArray[0] = 0.3;
+    fRDepPtCutArray[1] = 0.3;
+    fRDepPtCutArray[2] = 0.25;
+    fRDepPtCutArray[3] = 0.2;
+    fRDepPtCutArray[4] = 0.2;
+    fRDepPtCutArray[5] = 0.15;
+
+    break;
+  case 30: //u  R-Dep pT cut NomB Soft ;0.050 GeV + min gamma pT cut of 100 MeV
+    fSinglePtCut = 0.050;
+    fPtCut       = 0.1;
+    fDoRDepPtCut = kTRUE;
+    fRArray = new Double_t[fPtCutArraySize];
+    fRArray[0] = 13.;
+    fRArray[1] = 33.5;
+    fRArray[2] = 55.;
+    fRArray[3] = 72.;
+    fRArray[4] = 95.;
+    fRArray[5] = 180.;    
+
+    fRDepPtCutArray = new Double_t[fPtCutArraySize];
+    fRDepPtCutArray[0] = 0.25;
+    fRDepPtCutArray[1] = 0.25;
+    fRDepPtCutArray[2] = 0.2;
+    fRDepPtCutArray[3] = 0.2;
+    fRDepPtCutArray[4] = 0.15;
+    fRDepPtCutArray[5] = 0.1;
+    //    fRDepPtCutArray[6] = {0.25, 0.25, 0.2, 0.2, 0.15, 0.1};
+    break;
+  case 31: //v  R-Dep pT cut LowB Default 0.050 GeV + min gamma pT cut of 100 MeV
+    fSinglePtCut = 0.020;
+    fPtCut       = 0.1;
+    fDoRDepPtCut = kTRUE;
+    fRArray = new Double_t[fPtCutArraySize];
+    fRArray[0] = 13.;
+    fRArray[1] = 33.5;
+    fRArray[2] = 55.;
+    fRArray[3] = 72.;
+    fRArray[4] = 95.;
+    fRArray[5] = 180.;    
+
+    fRDepPtCutArray = new Double_t[fPtCutArraySize];
+    fRDepPtCutArray[0] = 0.2;
+    fRDepPtCutArray[1] = 0.2;
+    fRDepPtCutArray[2] = 0.15;
+    fRDepPtCutArray[3] = 0.15;
+    fRDepPtCutArray[4] = 0.15;
+    fRDepPtCutArray[5] = 0.15;
+    //    fRDepPtCutArray[6] = {0.2, 0.2, 0.15, 0.15, 0.15, 0.15};
+    break;
+  case 32: //w  R-Dep pT cut LowB Soft ;0.050 GeV + min gamma pT cut of 100 MeV
+    fSinglePtCut = 0.020;
+    fPtCut       = 0.1;
+    fDoRDepPtCut = kTRUE;
+    fRArray = new Double_t[fPtCutArraySize];
+    fRArray[0] = 13.;
+    fRArray[1] = 33.5;
+    fRArray[2] = 55.;
+    fRArray[3] = 72.;
+    fRArray[4] = 95.;
+    fRArray[5] = 180.;    
+    fRDepPtCutArray = new Double_t[fPtCutArraySize];
```

Comment:
Why not directly plugging in '6' for fPtCutArraySize? You made it possible to be set in the AddTask, but it is anyway hardcoded to be of dimension 6 in ConversionPhotonCuts.

Reply:
Would save you a lot of lines in this commit

Reply:
daniel, I am not sure what you exactly mean. What I am doing now is to change around L1684 , and put a loop
instead of the hard coded numbers. for the definition of the arrays within the cut, I do not know how to do it in one line